centos6 / gcc 4.4.7 compile error

```
    gcc -pthread -Werror=declaration-after-statement -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/vagrant/.pyenv/versions/3.4.0/include/python3.4m -c src/borg/hashindex.c -o build/temp.linux-x86_64-3.4/src/borg/hashindex.o
    In file included from src/borg/cache_sync/unpack.h:390,
                     from src/borg/cache_sync/cache_sync.c:19,
                     from src/borg/hashindex.c:437:
    src/borg/cache_sync/unpack_template.h: In function ‘unpack_execute’:
    src/borg/cache_sync/unpack_template.h:59: error: ISO C90 forbids mixed declarations and code
    error: command 'gcc' failed with exit status 1
```

--- a/src/borg/cache_sync/unpack_template.h
+++ b/src/borg/cache_sync/unpack_template.h
@@ -54,8 +54,6 @@ static inline void unpack_init(unpack_context* ctx)
 
 static inline int unpack_execute(unpack_context* ctx, const char* data, size_t len, size_t* off)
 {
-    assert(len >= *off);
-
     const unsigned char* p = (unsigned char*)data + *off;
     const unsigned char* const pe = (unsigned char*)data + len;
     const void* n = NULL;
@@ -69,6 +67,8 @@ static inline int unpack_execute(unpack_context* ctx, const char* data, size_t l
     unpack_stack* c = NULL;
 
     int ret;
+
+    assert(len >= *off);
 
 #define construct_cb(name) \
     construct && unpack_callback ## name

